### PR TITLE
make requiredPaymentDetails optional

### LINKED ISSUE #90: make `requiredPaymentDetails` optional
[this](https://github.com/TBD54566975/tbdex-js/blob/main/packages/protocol/src/types.ts#L112) should be optional as per the [spec](https://github.com/TBD54566975/tbdex/tree/main/specs/protocol#paymentmethod)

### PR DESCRIPTION
**Pull request by finn-tbd**
fixes #90

### CODE REVIEW

**Review by diehuxx — APPROVED** [review]

**Comment by finn-tbd** [comment]
left this in draft state because I wasn't sure if further checks were needed. It looks like this field is used in a `protoc/src/message-kinds/rfq.ts` where it is assumed to be set. Should we have some safety around checking there? I'm missing a lot of context around what this field is and how it may be used.

**Comment by diehuxx** [comment]
> It looks like this field is used in a protoc/src/message-kinds/rfq.ts where it is assumed to be set.

@finn-tbd Good catch. The fix for that would conflict with #141 (which will also make the TS compiler strict enough to catch errors like that in the future). Once #141 is merged, I can push a commit to this PR fixing `protoc/src/message-kinds/rfq.ts` is that's alright with you.

**Comment by finn-tbd** [comment]
sounds good

**Comment by diehuxx** [comment]
TODO:
- [ ] After https://github.com/TBD54566975/tbdex/pull/228 is merged, update tbdex submodule back to `main`.

**Review by phoebe-lew — APPROVED** [review]

**Comment by finn-tbd** [comment]
Blocked pending https://github.com/TBD54566975/tbdex/pull/227

File: packages/protocol/src/message-kinds/rfq.ts [hunk 1]
```diff
@@ -146,28 +145,40 @@ export class Rfq extends Message {
...
     for (const paymentMethodMatch of paymentMethodMatches) {
+      if (!paymentMethodMatch.requiredPaymentDetails) {
+        // If requiredPaymentDetails is omitted, and paymentDetails is also omitted, we have a match
+        if (!rfqPaymentMethod.paymentDetails) {
+          return
+        }
+
+        // paymentDetails is present even though requiredPaymentDetails is omitted. This is unsatisfactory.
+        invalidPaymentDetailsErrors.add(new Error('paymentDetails must be omitted when requiredPaymentDetails is omitted'))
+        continue
```
Comment (KendallWeihe) [inline packages/protocol/src/message-kinds/rfq.ts:166]: rather than a `continue` here, could we do an `else` block?

**Review by KendallWeihe — APPROVED** [review]
one comment added, but not strictly required (opinionated take)

feel free to merge once this blocking PR is merged https://github.com/TBD54566975/tbdex/pull/227

File: packages/protocol/src/message-kinds/rfq.ts (hunk 1 above)
Comment (diehuxx) [reply packages/protocol/src/message-kinds/rfq.ts]: works for me

**Pull request merged**